edit-user-role-page.test.tsx

```diff
@@ -0,0 +1,79 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import EditUserRolePage from '@/app/edit-user-role-page/page';
+import fetchMock from 'jest-fetch-mock';
+
+jest.mock('../../../components/UserCard', () => {
+    return function MockUserCard({ user }: { user: any }) {
+      return <div data-testid="user-card">{user.firstName} {user.lastName}</div>;
+    };
+  });
+  
+  jest.mock('../../../components/UnauthorizedPageMessage', () => {
+    return function MockUnauthorizedMessage() {
+      return <div data-testid="unauthorized-message">Unauthorized</div>;
+    };
+  });
+  
+
+describe('EditUserRolePage Component', () => {
+  beforeEach(() => {
+    fetchMock.resetMocks();
+    localStorage.clear();
+  });
+
+  test('renders User Management page for admin users and displays user info', async () => {
+    // Mock localStorage for admin role
+    const mockToken = JSON.stringify({
+      role: 'admin',
+    });
+    localStorage.setItem(
+      'token',
+      `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.${btoa(mockToken)}.signature`
+    );
+
+    // Mock user data fetch
+    const mockUsers = [

```

Comment:
If you need assistance connecting to your own local database instance, please let me know and I can walk you through it. We probably should remove this mockUsers and fetchMock this page should fetch directly from the database